lib/sidekiq/throttled/config.rb

```diff
@@ -39,6 +48,19 @@ def cooldown_threshold=(value)
 
         @cooldown_threshold = value
       end
+
+      # @!attribute [w] default_requeue_options
+      def default_requeue_options=(options)
+        requeue_with = options.delete(:with).intern || :enqueue
+
+        @default_requeue_options = options.merge({ with: requeue_with })
+      end
+
+      def reset!
+        @cooldown_period    = 1.0
+        @cooldown_threshold = 100
+        @default_requeue_options = { with: :enqueue }
+      end
```

Comment:
IMO it's better to introduce `#configure!` on Sidekiq::Throttled` to reconfigure if needed. As `#reset` will only work during `configure` anyway (config gets frozen)

Reply:
I'm not sure I follow you. I understand that this will only work if used as we do [here](https://github.com/ixti/sidekiq-throttled/pull/150/files#diff-c794909eeb67670b32f641d0876bf4784748b2ad5382dd6cdc883b416b089de7R66). 
What would be the difference that you propose between `Sidekiq::Throttled#configure!` and `Sidekiq::Throttled#configure`? I thought a `reset!` method on `Config` would be cleaner to set all settings back to their default values. Alternatively I think something like the following would work:
```
Sidekiq::Throttled.configure do |config|
  config.cooldown_period    = 1.0
  config.cooldown_threshold = 100
  config.default_requeue_options = { with: :enqueue }
end
```
The problem IMO would be that you'd leak the default values out of `Config`. Even if declared as constants, I assumed it'd be more correct to keep the defaults encapsulated on the config class.

Reply:
The problem is that `config` gets frozen after the `#configure` block. So you can't call `Sidekiq::Throttled.config.reset!`, thus with current API one will have to:

``` ruby
Sidekiq::Throttled.configure do |config|
  config.reset!

  ...
end
```

Versus:

``` ruby
Sidekiq::Throttled.configure! do |config|
  ...
end
```

Reply:
After thinking a bit, I think `config.reset!` is actually cleaner approach :D